packages/features/calendar-cache/calendar-cache.repository.ts

```diff
@@ -118,13 +136,30 @@ export class CalendarCacheRepository implements ICalendarCacheRepository {
         },
       });
     } else {
+      const staleRecord = await prisma.calendarCache.findFirst({
+        where: {
+          credentialId,
+          key,
+          expiresAt: { gte: new Date(Date.now()) },
+          stale: true,
+        },
+      });
+
+      if (staleRecord) {
+        log.warn(
+          "Found stale cache record, not returning as valid cache",
+          safeStringify({ credentialId, key })
+        );
+      }
+
       cached = await prisma.calendarCache.findUnique({
         where: {
           credentialId_key: {
             credentialId,
             key,
           },
           expiresAt: { gte: new Date(Date.now()) },
+          stale: false,
         },
       });
     }
```

Comment:
DevinAI, instead of doble querying. remove the stale from the where filters. And before doing 

return cached

Do a conditional:

if (cached.stale) return null;

Also use the current log.info to log if we're serving a stale cache or not. Also if there's no actual hit, log: skipping availablity... etc.